pyLemur/lin_alg_wrappers.py

```diff
@@ -18,18 +18,27 @@ def fit_pca(Y, n, center=True):
         The number of principal components to return.
     center : bool, default=True
         If True, the data will be centered before computing the covariance matrix.
+    device : str, default="cpu"
     Returns
     -------
     pca : sklearn.decomposition.PCA
         The PCA object.
     """
     if center:
-        pca = skd.PCA(n_components=n)
+        if device == "gpu":
+            from cuml import PCA as cumlPCA
+            pca = cumlPCA(n_components=n,output_type ="numpy")
+        else:
+            pca = skd.PCA(n_components=n)
         emb = pca.fit_transform(Y)
         coord_system = pca.components_
         mean = pca.mean_
     else:
-        svd = skd.TruncatedSVD(n_components=n, algorithm='arpack')
+        if device == "gpu":
+            from cuml import TruncatedSVD as cumlTSVD
+            svd = cumlTSVD(n_components=n,output_type ="numpy")
```

Comment:
I was just looking at the documentation for [cuml.TruncatedSVD](https://docs.rapids.ai/api/cuml/stable/api/#truncated-svd) and saw that it uses the full decomposition by default. Does that mean that it internally creates the full $gene \times gene$ matrix? Have you, by any chance, looked at how much less accurate `method = "Jacobi"` is?

Reply:
I'll set both PCA and TSVD to Jacobi as the algo